oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/userinfo/DefaultOAuth2UserService.java

```diff
@@ -127,7 +135,11 @@ public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2Authentic
 		}
 
 		Map<String, Object> userAttributes = response.getBody();
-		Set<GrantedAuthority> authorities = Collections.singleton(new OAuth2UserAuthority(userAttributes));
+		Set<GrantedAuthority> authorities = new LinkedHashSet<>();
+		authorities.add(new OAuth2UserAuthority(userAttributes));
+		for (String authority : getAuthorities(() -> userAttributes)) {
+			authorities.add(new SimpleGrantedAuthority("SCOPE_" + authority));
```

Comment:
I don't think the logic from `JwtGrantedAuthoritiesConverter` should be replicated here. I believe the additional default authorities can be simplified like this:

Current authority: `OAuth2UserAuthority` -> `ROLE_USER`

Additional authorities: `OAuth2UserRequest.OAuth2AccessToken.scopes` e.g. `SCOPE_profile`, `SCOPE_email`

** The assumption is that `OAuth2UserRequest.OAuth2AccessToken.scopes` contains `profile` and `email`, meaning the user authorized the client to access those scopes during the authorization approval step. NOTE: These scopes would be initially configured with the `ClientRegistration.scopes`.